dask/dataframe/csv.py

```diff
@@ -182,6 +189,15 @@ def read_csv(urlpath, blocksize=2**25, chunkbytes=None,
     if isinstance(kwargs.get('header'), list):
         raise TypeError("List of header rows not supported for dd.read_csv")
 
+    if blocksize == 'auto':
+        if psutil is not None:  # guess blocksize if psutil is installed
+            total_mem = psutil.virtual_memory().total
+            cpu_count = psutil.cpu_count()
+            blocksize = round(total_mem / cpu_count / 10)
+            blocksize = min(blocksize, int(64e6))
+        else:
+            blocksize = 2**25  # or use default one if not
```

Comment:
Done that. There are still two monkeypatches for integration testing. Hope it's ok. :)